src/south/ListenProtocolHandler.class.js

```diff
@@ -0,0 +1,51 @@
+const ProtocolHandler = require('./ProtocolHandler.class')
+
+/**
+ * Class ListenProtocolHandlerClass: provides general attributes and methods for "Listen" type South Protocols.
+ * Building a new "Listen" type South Protocol means to extend this class, and to surcharge
+ * the following methods:
+ * - **listen**: A special scanMode can be created for a protocol (for example MQTT). In this configuration, the
+ * driver will be able to "listen" for updated values.
+ * - **connect**: to allow to establish proper connection to the data source(optional)
+ * - **disconnect**: to allow proper disconnection (optional)
+ * In addition, it is possible to use a number of helper functions:
+ *
+ * All other operations (cache, store&forward, communication to North applications) will be
+ * handled by the OIBus engine and should not be taken care at the South level.
+ *
+ */
+class ListenProtocolHandler extends ProtocolHandler {
```

Comment:
I thought we would not create another parent class. I think it may limit the south developers. For example,  how do create a south that can support both listening to points and reading to the last point? (such as OPCUA)

Reply:
We can keep one single parent class, but it will be difficult to know which methods should be overwritten when implementing a new South.
For OPCUA we have separate implementations for OPCUA_DA and OPCUA_HA. If we want to implement a LastPoint OPCUA we could create a third OPCUA South.
Mixing different functionalities makes harder to understand and maintain the code.

Reply:
on the other hand, every "base" improvement we do in UA (for example, new method of identification), have to be replicated in mulitple places.

Reply:
I agree that OPCUA_DA should have both (lastPoint as it is for now, and listen that will be developed later). In this case, JF suggests to check in the parent class if the function is implemented.

The constructor could be something like : 
class MQTT extends ProtocolHandler {
constructor ()
{
super({supportListen: true, supportLastPoint: false, supportHistory: false})

And the check in the parent class could be like that : 
if (listenSupport) if (typeof listen !== “function”) ... log erreur

Reply:
Since we define listen() in ProtocolHandler (logging that it should be overwritten by the South implementation) it will always be a function and this check doesn't help us.

Reply:
unless we remove the default listen() from ProtocolHandler.

Reply:
So do you want to remove those the listen(), lastPointQuery() and historyQuery() methods from ProtocolHandler and just document it somewhere that a new South implementation must implement some of them (based on the what type of South it is), and this way we could add the check if the required method is implemented?
I'm not sure yet, if this.listen is already defined when calling the parent constructor, but I will check it.